<ui version="4.0" >
 <author>GeometryFactory</author>
 
 <class>Boolean_operations_2</class>
 
 <widget class="QMainWindow" name="Boolean_operations_2" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>Boolean operations on polygons</string>
  </property>
  <property name="windowIcon" >
   <iconset resource="../resources/CGAL.qrc" >
    <normaloff>:/cgal/logos/cgal_icon</normaloff>:/cgal/logos/cgal_icon</iconset>
  </property>
  
  <widget class="QWidget" name="centralwidget" >
   <layout class="QGridLayout" >
    <item row="0" column="0" >
     <widget class="QGraphicsView" name="graphicsView" >
      <property name="focusPolicy" >
       <enum>Qt::StrongFocus</enum>
      </property>
      <property name="verticalScrollBarPolicy" >
       <enum>Qt::ScrollBarAlwaysOn</enum>
      </property>
      <property name="horizontalScrollBarPolicy" >
       <enum>Qt::ScrollBarAlwaysOn</enum>
      </property>
      <property name="transformationAnchor" >
       <enum>QGraphicsView::NoAnchor</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  
  <widget class="QMenuBar" name="menubar" >
   <property name="geometry" >
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile" >
    <property name="title" >
     <string>&amp;File</string>
    </property>
    <addaction name="actionNew" />
    <addaction name="actionNewWindow" />
    <addaction name="separator" />
    <addaction name="actionOpenLinear" />
    <addaction name="actionOpenDXF" />
    <addaction name="actionOpenBezier" />
    <addaction name="separator" />
    <addaction name="actionPrint" />
    <addaction name="separator" />
    <addaction name="actionClose" />
    <addaction name="actionQuit" />
   </widget>
   <widget class="QMenu" name="menuHelp" >
    <property name="title" >
     <string>&amp;Help</string>
    </property>
    <addaction name="actionHowTo" />
    <addaction name="separator" />
    <addaction name="actionAbout" />
    <addaction name="actionAboutCGAL" />
   </widget>
   <addaction name="menuFile" />
   <addaction name="menuHelp" />
  </widget>

  <widget class="QToolBar" name="viewToolBar" >
   <property name="windowTitle" >
    <string>Visualization Tools</string>
   </property>
   <attribute name="toolBarArea" >
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak" >
    <bool>false</bool>
   </attribute>
   <addaction name="actionUndo" />
   <addaction name="actionRedo" />
  </widget>
  
  <widget class="QToolBar" name="inputToolBar" >
   <property name="windowTitle" >
    <string>Input Tools</string>
   </property>
   <attribute name="toolBarArea" >
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak" >
    <bool>false</bool>
   </attribute>
   <widget class="QRadioButton" name="redActive">
    <property name="text">
     <string>Red</string>
    </property>
    <property name="checked" >
      <bool>true</bool>
    </property>
   </widget>
   <widget class="QRadioButton" name="blueActive">
    <property name="text">
     <string>Blue</string>
    </property>
    <property name="checked" >
      <bool>false</bool>
    </property>
   </widget>
   <addaction name="actionInsertPolygon" />
   <addaction name="actionInsertCircle" />
   <addaction name="actionInsertBezier" />
  </widget>
  
  <widget class="QToolBar" name="operationsToolBar" >
   <property name="windowTitle" >
    <string>Operation Tools</string>
   </property>
   <attribute name="toolBarArea" >
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak" >
    <bool>false</bool>
   </attribute>
   <addaction name="actionIntersection" />
   <addaction name="actionUnion" />
   <addaction name="actionBlueMinusRed" />
   <addaction name="actionRedMinusBlue" />
   <addaction name="actionSymmDiff" />
   <addaction name="actionMinkowskiSum" />
   <addaction name="actionBlueComplement" />
   <addaction name="actionRedComplement" />
   <addaction name="actionAllBlue" />
   <addaction name="actionAllRed" />
   <addaction name="actionDeleteBlue" />
   <addaction name="actionDeleteRed" />
   <addaction name="actionRefresh" />
  </widget>
  
  <widget class="QStatusBar" name="statusbar" >
  </widget>
  
  <action name="actionNew" >
   <property name="text" >
    <string>&amp;New</string>
   </property>
  </action>
  
  <action name="actionNewWindow" >
   <property name="text" >
    <string>New &amp;Window</string>
   </property>
  </action>
  
  <action name="actionOpenLinear" >
   <property name="text" >
    <string>&amp;Open Linear Polygon</string>
   </property>
  </action>
  
  <action name="actionOpenDXF" >
   <property name="text" >
    <string>&amp;Open DXF</string>
   </property>
  </action>
  
  <action name="actionOpenBezier" >
   <property name="text" >
    <string>&amp;Open Bezier Polygon</string>
   </property>
  </action>

  <action name="actionPrint" >
   <property name="text" >
    <string>&amp;Print</string>
   </property>
  </action>
  
  <action name="actionClose" >
   <property name="text" >
    <string>&amp;Close</string>
   </property>
   <property name="shortcut" >
    <string>Ctrl+X</string>
   </property>
  </action>
  
  <action name="actionQuit" >
   <property name="text" >
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut" >
    <string>Ctrl+Q</string>
   </property>
  </action>
  
  <action name="actionHowTo" >
   <property name="text" >
    <string>How To</string>
   </property>
   <property name="shortcut" >
    <string>F1</string>
   </property>
  </action>
  
  <action name="actionAbout" >
   <property name="text" >
    <string>&amp;About</string>
   </property>
  </action>
  
  <action name="actionAboutCGAL" >
   <property name="text" >
    <string>About &amp;CGAL</string>
   </property>
  </action>
  

  <action name="actionUndo" >
   <property name="text" >
    <string>Undo</string>
   </property>
   <property name="toolTip" >
    <string>History Back</string>
   </property>
  </action>

  <action name="actionRedo" >
   <property name="text" >
    <string>Redo</string>
   </property>
   <property name="toolTip" >
    <string>History Fordward</string>
   </property>
  </action>
  
  <action name="actionMakeBlueActive" >
   <property name="text" >
    <string>Blue</string>
   </property>
  </action>
  
  <action name="actionMakeRedActive" >
   <property name="text" >
    <string>Red</string>
   </property>
  </action>
  
  <action name="actionInsertPolygon" >
   <property name="checkable" >
    <bool>true</bool>
   </property>
   <property name="checked" >
    <bool>false</bool>
   </property>
   <property name="icon" >
    <iconset resource="boolean_operations_2.qrc" >
     <normaloff>:/cgal/Actions/icons/insert_polygon.xpm</normaloff>:/cgal/Actions/icons/insert_polygon.xpm
    </iconset>
   </property>
   <property name="toolTip" >
    <string>Insert Circular Polygon</string>
   </property>
  </action>
  
  <action name="actionInsertCircle" >
   <property name="checkable" >
    <bool>true</bool>
   </property>
   <property name="checked" >
    <bool>false</bool>
   </property>
   <property name="icon" >
    <iconset resource="boolean_operations_2.qrc" >
     <normaloff>:/cgal/Actions/icons/insert_circle.xpm</normaloff>:/cgal/Actions/icons/insert_circle.xpm
    </iconset>
   </property>
   <property name="toolTip" >
    <string>Insert Circule</string>
   </property>
  </action>
  
  <action name="actionInsertBezier" >
   <property name="checkable" >
    <bool>true</bool>
   </property>
   <property name="checked" >
    <bool>false</bool>
   </property>
   <property name="icon" >
    <iconset resource="boolean_operations_2.qrc" >
     <normaloff>:/cgal/Actions/icons/insert_circle.xpm</normaloff>:/cgal/Actions/icons/insert_circle.xpm
    </iconset>
   </property>
   <property name="toolTip" >
    <string>Insert Bezier</string>
   </property>
  </action>
  
  <action name="actionIntersection" >
   <property name="icon" >
    <iconset resource="boolean_operations_2.qrc" >
     <normaloff>:/cgal/Actions/icons/intersection.xpm</normaloff>:/cgal/Actions/icons/intersection.xpm
    </iconset>
   </property>
   <property name="toolTip" >
    <string>Intersection</string>
   </property>
  </action>

  <action name="actionUnion" >
   <property name="icon" >
    <iconset resource="boolean_operations_2.qrc" >
     <normaloff>:/cgal/Actions/icons/union.xpm</normaloff>:/cgal/Actions/icons/union.xpm
    </iconset>
   </property>
   <property name="toolTip" >
    <string>Union</string>
   </property>
  </action>

  <action name="actionBlueMinusRed" >
   <property name="icon" >
    <iconset resource="boolean_operations_2.qrc" >
     <normaloff>:/cgal/Actions/icons/diff_PQ.xpm</normaloff>:/cgal/Actions/icons/diff_PQ.xpm
    </iconset>
   </property>
   <property name="toolTip" >
    <string>Difference between Blue and Red</string>
   </property>
  </action>
 
  <action name="actionRedMinusBlue" >
   <property name="icon" >
    <iconset resource="boolean_operations_2.qrc" >
     <normaloff>:/cgal/Actions/icons/diff_QP.xpm</normaloff>:/cgal/Actions/icons/diff_QP.xpm
    </iconset>
   </property>
   <property name="toolTip" >
    <string>Difference between Red and Blue</string>
   </property>
  </action>

  <action name="actionSymmDiff" >
   <property name="icon" >
    <iconset resource="boolean_operations_2.qrc" >
     <normaloff>:/cgal/Actions/icons/symm_diff.xpm</normaloff>:/cgal/Actions/icons/symm_diff.xpm
    </iconset>
   </property>
   <property name="toolTip" >
    <string>Symmetric Difference</string>
   </property>
  </action>
  
  <action name="actionMinkowskiSum" >
   <property name="icon" >
    <iconset resource="boolean_operations_2.qrc" >
     <normaloff>:/cgal/Actions/icons/mink_sum.xpm</normaloff>:/cgal/Actions/icons/mink_sum.xpm
    </iconset>
   </property>
   <property name="toolTip" >
    <string>Minkowski Sum</string>
   </property>
  </action>
  
  <action name="actionBlueComplement" >
   <property name="icon" >
    <iconset resource="boolean_operations_2.qrc" >
     <normaloff>:/cgal/Actions/icons/comp_P.xpm</normaloff>:/cgal/Actions/icons/comp_P.xpm
    </iconset>
   </property>
   <property name="toolTip" >
    <string>Blue Complement</string>
   </property>
  </action>
  
  <action name="actionRedComplement" >
   <property name="icon" >
    <iconset resource="boolean_operations_2.qrc" >
     <normaloff>:/cgal/Actions/icons/comp_Q.xpm</normaloff>:/cgal/Actions/icons/comp_Q.xpm
    </iconset>
   </property>
   <property name="toolTip" >
    <string>Red Complement</string>
   </property>
  </action>
  
  <action name="actionAllBlue" >
   <property name="icon" >
    <iconset resource="boolean_operations_2.qrc" >
     <normaloff>:/cgal/Actions/icons/make_P.xpm</normaloff>:/cgal/Actions/icons/make_P.xpm
    </iconset>
   </property>
   <property name="toolTip" >
    <string>Make result Blue</string>
   </property>
  </action>
  
  <action name="actionAllRed" >
   <property name="icon" >
    <iconset resource="boolean_operations_2.qrc" >
     <normaloff>:/cgal/Actions/icons/make_Q.xpm</normaloff>:/cgal/Actions/icons/make_Q.xpm
    </iconset>
   </property>
   <property name="toolTip" >
    <string>Make result Red</string>
   </property>
  </action>
  
  <action name="actionDeleteBlue" >
   <property name="icon" >
    <iconset resource="boolean_operations_2.qrc" >
     <normaloff>:/cgal/Actions/icons/del_P.xpm</normaloff>:/cgal/Actions/icons/del_P.xpm
    </iconset>
   </property>
   <property name="toolTip" >
    <string>Delete Blue</string>
   </property>
  </action>
  
  <action name="actionDeleteRed" >
   <property name="icon" >
    <iconset resource="boolean_operations_2.qrc" >
     <normaloff>:/cgal/Actions/icons/del_Q.xpm</normaloff>:/cgal/Actions/icons/del_Q.xpm
    </iconset>
   </property>
   <property name="toolTip" >
    <string>Delete Red</string>
   </property>
  </action>
  
  <action name="actionRefresh" >
   <property name="icon" >
    <iconset resource="boolean_operations_2.qrc" >
     <normaloff>:/cgal/Actions/icons/refresh.xpm</normaloff>:/cgal/Actions/icons/refresh.xpm
    </iconset>
   </property>
   <property name="toolTip" >
    <string>Refresh</string>
   </property>
  </action>
  
 </widget>
 
 <connections>
  <connection>
   <sender>redActive</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>on_actionMakeRedActive_toggled(bool)</slot>
  </connection>
  <connection>
   <sender>blueActive</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>on_actionMakeBlueActive_toggled(bool)</slot>
  </connection>
 </connections>
 <slots>
   <slot>on_actionMakeRedActive_toggled(bool)</slot>
   <slot>on_actionMakeBlueActive_toggled(bool)</slot>
 </slots>
 
 <resources>
  <include location="boolean_operations_2.qrc" />
 </resources>
 
</ui>
